front-end/src/components/LoginForm.jsx

```diff
@@ -36,26 +36,16 @@ export default class LoginForm extends React.Component {
     axios.post(process.env.REACT_APP_API_URL + "/api/rest-auth/login/", {
       username: this.state.username,
       password: this.state.password
-    }).then(res => {
-      //Store token in a cookie
-      // document.cookie = "accessToken=" + res.data.key;
-
-      axios.get(process.env.REACT_APP_API_URL + `/api/auth-user/${this.state.username}/`
-      ).then(res => {
-        this.props.setLoggedInUser({username: this.state.username, password: this.state.password, authorId: res.data.id});
-        this.props.history.push("/");
-      }).catch(err => {
-        this.setState({loginErr: true});
-      });
-
-      axios.get(process.env.REACT_APP_API_URL + "/api/nodes/").then(res => {
-        this.props.setNodes(res.data);
-      }).catch(err => {
-        this.setState({loginErr: true});
-      });
-
+    }).then(_ => {
+      return axios.get(process.env.REACT_APP_API_URL + "/api/nodes/");
```

Comment:
Off-topic: Is there a convention guide for this style of promise chaining? Personally, I find this a bit harder to read as opposed to nested callbacks, since the `return` statements aren't breaking out of the function. I might be in the minority though

We should leave this as is, since either way I'm good with it, but if I'm writing code that's not idiomatic I'd like to improve in that area

Reply:
Good question, I honestly don't know what is the usual practice is in JS/TS, but I usually try to stay away from having too many levels of indentation.

Reply:
https://www.freecodecamp.org/news/how-to-deal-with-nested-callbacks-and-avoid-callback-hell-1bc8dc4a2012/
Yeah nesting callbacks definitely isn't great after a while, you bring up a good point :+1: . I'll have to chew some stuff over in my head to figure out what the nicest way of doing promises are. In the meantime I'll strive towards `.then` chaining

Reply:
Sounds good!